client/me/help/help-contact-form/index.jsx

```diff
@@ -175,6 +197,22 @@ export class HelpContactForm extends React.PureComponent {
 
 	getSibylQuery = () => ( this.state.subject + ' ' + this.state.message ).trim();
 
+	doRequestSite = () => {
+		if ( resemblesUrl( this.state.userDeclaredUrl ) ) {
+			// The API would reject something like "https://wp.com/slug" - it'd need to be "wp.com".
+			const url = this.state.userDeclaredUrl;
+			const query = url.includes( '://' )
+				? new URL( this.state.userDeclaredUrl ).hostname
+				: new URL( 'http://' + this.state.userDeclaredUrl ).hostname;
+
+			this.props.requestSite( query, false, true ).then( ( siteData ) =>
+				this.setState( {
+					siteData,
+				} )
```

Comment:
In case of success, the `siteData` will be `undefined`. The promise returned from `requestSite` doesn't provide the site data.

Reply:
I'll check this out again today, but from what I remember, I don't believe that there should be successful cases because that would require the user having capabilities on that site. In order for this to be called, the user needs to select a box that declares they need help with a site which they don't have capabilities on.

Reply:
Oh I see, in this case, we're pretty much guaranteed to see a site that is not managed by us, and get the `SITE_REQUEST_FAILURE` action returned in this case.

I tried to make `requestSite` returned promises more reasonable in #53187. If that gets landed, the `HelpContactForm` can install `.then( site => ... )` and `.catch( error => ... )` and set its local state accordingly.

Reply:
>  In order for this to be called, the user needs to select a box that declares they need help with a site which they don't have capabilities on.

This is over my head so I'm not sure I understand the implications, but I do want to point out people will frequently ask for help with a site they're not connected to (like if they are logged into the [wrong account](https://wordpress.com/forums/topic-tag/other-user/), or are struggling with [account recovery](https://wordpress.com/forums/topic-tag/account-recovery/)).

Ultimately I hope folks can share a site link even if they are in the wrong account, and even if it's .org (or wix for that matter), but that they'll have a heads up.

Reply:
> Ultimately I hope folks can share a site link even if they are in the wrong account, and even if it's .org (or wix for that matter), but that they'll have a heads up.

Just to be clear, both of these should happen, and the site link will still be included! :)

Reply:
Awesome. THANK YOU!